acme/writer.go

```diff
@@ -255,18 +256,23 @@ func testAddresses(ctx context.Context, p peer.ID, addrs []string) error {
 // libp2p clients can set agent version to arbitrary strings,
 // and the metric labels have to have a bound cardinality
 func agentType(agentVersion string) string {
-	if strings.HasPrefix(agentVersion, "kubo/") {
+	switch {
+	case strings.HasPrefix(agentVersion, "kubo/"):
 		return "kubo"
-	}
-	if strings.HasPrefix(agentVersion, "helia/") {
+	case strings.HasPrefix(agentVersion, "helia/"):
 		return "helia"
-	}
-	// TODO:  revisit once js0libp2p cleans up default user agents to something unique and not "libp2p/"
-	if strings.HasPrefix(agentVersion, "libp2p/") || strings.HasPrefix(agentVersion, "js-libp2p/") {
+	case strings.HasPrefix(agentVersion, "libp2p/") || strings.HasPrefix(agentVersion, "js-libp2p/"): // TODO:  revisit once js0libp2p cleans up default user agents to something unique and not "libp2p/"
```

Comment:
@achingbrain do you remember how does HTTP User-Agent and libp2p identify's agentVersion look like in requests made by https://www.npmjs.com/package/@libp2p/auto-tls?

We use libp2p one on success, but want to fallback to User-Agent from http request if we could not do libp2p identify, and would like this PR to correctly identify @libp2p/auto-tls  in both success and error scenarios.

Reply:
Filled https://github.com/ipshipyard/p2p-forge/issues/48 to figure it out without blocking this PR.

Reply:
We don't specify anything right now so it would probably fall back to the `acme-client` [default settings](https://github.com/publishlab/node-acme-client/blob/master/src/axios.js#L19).

Something like `node-acme-client/5.4.0`.  It would be better if it logged the js-libp2p version too.

Reply:
Or do you mean setting up the acme challenge response?  Again, we just use the default so I think it just ends up as `"node"` currently.

Reply:
https://github.com/libp2p/js-libp2p/pull/2932 + this PR will do the trick, thanks!

Reply:
I'd think about maybe doing a `contains` here rather than `hasPrefix`, the libp2p agent string is:

`$libp2pVersion $platformVersion`,

Helia does

`$heliaVersion $libp2pVersion $platformVersion`